ciao-controller/internal/datastore/datastore.go

```diff
@@ -2280,3 +2257,61 @@ func (ds *Datastore) UnMapExternalIP(address string) error {
 
 	return nil
 }
+
+func (ds *Datastore) generateCNCIWorkload() {
+	// generate the CNCI workload.
+	config := `---
+#cloud-config
+users:
+  - name: cloud-admin
+    gecos: CIAO Cloud Admin
+    lock-passwd: false
+    passwd: $6$rounds=4096$w9I3hR4g/hu$AnYjaC2DfznbPSG3vxsgtgAS4mJwWBkcR74Y/KHNB5OsfAlA4gpU5j6CHWMOkkt9j.9d7OYJXJ4icXHzKXTAO.
+    sudo: ALL=(ALL) NOPASSWD:ALL
+...
+`
+	cpus := payloads.RequestedResource{
+		Type:      payloads.VCPUs,
+		Value:     4,
+		Mandatory: false,
+	}
+
+	mem := payloads.RequestedResource{
+		Type:      payloads.MemMB,
+		Value:     128,
```

Comment:
I could do this 2 ways - one way is to add a flag available only to the privileged user to be able to create CNCI workloads. The disadvantage of this approach is that if the admin messes up the CNCI workload definition, his cluster is hosed. Another option is to make parts of the cnci configurable via a separate api (such as /cncis) and have only certain values be configurable (like mem or disk for example). this has the disadvantage of needing yet another interface, but reduces the likelyhood of screwing up the cnci definition.